File: tests/test_solver_errorfcn.py
import numpy as np

from akiles2d.simrc import simrc
from akiles2d.solver import errorfcn


def test_errorfcn_length():
  data = simrc()
  data.ions.mu = 10
  solution = {
    "h": np.array([1.0, 2.0]),
    "r": np.array([0.0, 0.0]),
    "phi": np.array([0.0, -0.5]),
    "ne00p": data.guess.ne00p,
    "npoints": 2,
  }
  err = errorfcn(data, solution)
  assert err.shape[0] == solution["npoints"]

File: matlab_port/tests_matlab/test_solver_errorfcn.m
function tests = test_solver_errorfcn
% Migration tests for solver error function.
tests = functiontests(localfunctions);

function testErrorVectorLength(testCase)
  data = akiles2d.simrc();
  data.ions.mu = 10;
  solution.h = [1; 2];
  solution.r = [0; 0];
  solution.phi = [0; -0.5];
  solution.ne00p = data.guess.ne00p;
  solution.npoints = numel(solution.h);
  err = akiles2d.solver.errorfcn(data,solution);
  verifyEqual(testCase,numel(err),solution.npoints);
end

end
